proxy/envoy/discovery_test.go

```diff
@@ -83,6 +83,14 @@ func makeDiscoveryRequest(ds *DiscoveryService, method, url string, t *testing.T
 	return body
 }
 
+func commonSetup(t *testing.T) (*proxyconfig.MeshConfig, model.ConfigStore, *DiscoveryService) {
+	mesh := makeMeshConfig()
+	registry := memory.Make(model.IstioConfigTypes)
+	addConfig(registry, egressRule, t)
```

Comment:
please remove egressRule from common setup. The tests should test each rule/config in isolation.